modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/info_box/modal/InfoBoxModalPaymentMethodInput.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import {ClayRadio, ClayRadioGroup} from '@clayui/form';
+import {CommerceServiceProvider} from 'commerce-frontend-js';
+import React, {useEffect, useState} from 'react';
+
+const InfoBoxModalPaymentMethodInput = ({
+	inputValue,
+	orderId,
+	setInputValue,
+	setIsValid,
+	setParseResponse,
+	spritemap,
+}) => {
+	const [hasPaymentMethods, setHasPaymentMethods] = useState(false);
+	const [paymentMethods, setPaymentMethods] = useState([]);
+
+	useEffect(() => {
+		setParseResponse(() => (field, response) => {
+			return response['paymentMethodLabel'];
+		});
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	useEffect(() => {
+		CommerceServiceProvider.DeliveryCartAPI('v1')
+			.getCartPaymentMethodsPage(orderId)
+			.then(({items}) => {
+				const paymentMethodsAvailable = !!items.length;
+
+				setHasPaymentMethods(paymentMethodsAvailable !== undefined);
+				setIsValid(paymentMethodsAvailable !== undefined);
```

Comment:
in your case you cannot do `!== undefined` because you have always true or false (never undefined)
You have just to do
```
setHasPaymentMethods(paymentMethodsAvailable);
setIsValid(paymentMethodsAvailable);
```